### PR TITLE
Solving gravity and negative values problems

### PR DESCRIPTION
**Pull request by maycowmeira**
Used Math.abs() to get the absolute value of the AXIS, so it will never be negative anymore.
Gravity problem solved, instead of using TYPE_ACCELEROMETER, I used TYPE_LINEAR_ACCELERATION. What gives us the data without the gravity already.

### CODE REVIEW

File: dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java [hunk 1]
```diff
@@ -179,18 +179,18 @@ public void onSensorChanged(SensorEvent event) {
                     lastTimeCheck = currentTimeMilliseconds;
 
                     JSONObject accelerationJson = new JSONObject();
-
+                    //TODO: take the gravity away here
```
Comment (mike-mh) [inline dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java]: Remove comment

**Comment by mike-mh** [comment]
This solution doesn't work on Android 4.2 nor 4.4 devices. 

Just tested it on the watch and my own device. No acceleration data is read and stack trace says:

`03-01 10:06:54.115 31582-31582/com.example.michael.mqtttest E/SensorManager: sensor or listener is null`

The problem is `TYPE_LINEAR_ACCELERATION` is not defined until API level 20. We need to use code from API level 17 and below.

Sample found here:

http://developer.android.com/guide/topics/sensors/sensors_motion.html

Look at the 'Using the Accelerometer' section.

Also some good references here:

http://stackoverflow.com/questions/3377288/how-to-remove-gravity-factor-from-accelerometer-readings-in-android-3-axis-accel

**Comment by krxsky** [comment]
@0x4d464d48 Just curious, the doc says that `TYPE_LINEAR_ACCELERATION` is added in API 9. http://developer.android.com/reference/android/hardware/Sensor.html#TYPE_LINEAR_ACCELERATION

**Comment by mike-mh** [comment]
This looks good to me. Will merge after QA testing is complete.

**Comment by krxsky** [comment]
Also looks good to me. This low-pass filter thing is pretty interesting. Learned some new stuff :)

**Comment by mike-mh** [comment]
I just caught this. Michael, before Carlos QAs this branch, we need you to compile the project and then place the new .apk file into the `dcapp_apk` directory.

Go to the `dcapp` directory in a terminal and run:

```
./gradlew assembleDebug
```

After this is get the `app-debug.apk` file from the directory:

```
app/build/outputs/apk
```

Move `app-edbug.apk` it to the `dcapp_apk` directory

**Comment by mike-mh** [comment]
QA this sucker.

**Comment by CarlosBalladares** [comment]
An update on this, test are being run on this. Trying to figure out some how to modify some private fields to see to be able to run this as a test.

**Comment by CarlosBalladares** [comment]
Looks good. Merging

**Pull request merged**